scripts/server.py

```diff
@@ -550,66 +573,329 @@ def handle_client(conn, addr):
+                                                          headers={'Content-Type': 'application/octet-stream',
+                                                                   'Content-Encoding': 'gzip'},
+                                                          timeout=bw_default_post_timeout)
+                # Raise a error if response is not 2XX
+                soundscape_async_response.raise_for_status()
+
+                # Spit out the whole dict response if debugging
+                if debug_birdweather_submissions:
+                    print(f'BirdWeather Submission:: DEBUG:: Soundscape POST - RESPONSE: {soundscape_async_response}', flush=True)
+
+                # Extract some data
+                soundscape_response_json = soundscape_async_response.json()
+                # Get the soundscape id for the soundscape uploaded
+                soundscape_id = soundscape_response_json['soundscape']['id']
+
+                # Cache the soundscape id for corresponding to the filename the wave uploaded
+                birdweather_soundscape_id_cache('add', soundscape_filename, soundscape_id)
+
+                print(f"BirdWeather Submission:: Soundscape Successfully Uploaded - status:{soundscape_async_response.status_code} soundscape_id:{soundscape_id}", flush=True)
+            else:
+                # We found the soundscape filename and the Birdweather Soundscape ID for it, soundscape considered uploaded & use the ID for the detection
+                soundscape_id = find_existing_soundscape['soundscape_id']
+                if debug_birdweather_submissions:
+                    print(f"BirdWeather Submission:: Found Existing Soundscape in cache for {soundscape_filename} - using soundscape_id:{soundscape_id}", flush=True)
+
+            # Break the loop, if we reach here then were no exceptions and the detection posted successfully
+            break
+        except (requests.exceptions.ConnectionError, requests.exceptions.ConnectTimeout, requests.exceptions.ReadTimeout) as conn_exec:
+            # can_retry = True
+            print(f"BirdWeather Submission Error:: Soundscape POST - Connection Error! - {conn_exec}", flush=True)
+            continue
+        except requests.exceptions.RequestException as request_exc:
+            # Check if the status code is one that we can try on
+            req_status_code = request_exc.response.status_code
+            if req_status_code in bw_request_status_retry_for:
+
+                if debug_birdweather_submissions:
+                    extra_debug_output = f' - {request_exc.response.reason} - {request_exc}'
+
+                print(f"BirdWeather Submission Error:: Soundscape POST - HTTP Request Exception! {extra_debug_output}", flush=True)
+                continue
+            else:
+                print(f"BirdWeather Submission Error:: Soundscape POST - HTTP Request Exception! Cannot retry - {request_exc}", flush=True)
+                # break the loop on non-retryable status
```

Comment:
Not sure if that's a word, but if it is, it's spelled "retriable" ;)

Reply:
Yeah it's definitely not a common place compound word but gets used here and there, I've mostly seen it around Microsoft Azure AD SSO integration and SQL Server errors.

Either "retryable" or "retriable" are correct spelling for basically the same word, though either way might be suitable depending on the context.